int/codec.ts

```diff
@@ -38,25 +38,22 @@ export const i64 = new NumCodec(8, "BigInt64");
 
 // https://github.com/unstoppablejs/unstoppablejs/blob/7022e34f756ccc25e6ed9d4680284455b2ff714b/packages/scale-ts/src/codecs/fixed-width-ints.ts#L59-L74
 class X128Codec extends Codec<bigint> {
-  constructor(signed: boolean) {
-    super(
-      () => {
-        return 16;
-      },
-      (cursor, value) => {
-        cursor.view.setBigInt64(cursor.i, value, true);
-        cursor.view.setBigInt64(cursor.i + 8, value >> 64n, true);
-        cursor.i += 16;
-      },
-      (cursor) => {
-        const right = cursor.view.getBigUint64(cursor.i, true);
-        const left = cursor.view[
-          signed ? "getBigInt64" : "getBigUint64"
-        ](cursor.i + 8, true);
-        cursor.i += 16;
-        return (left << 64n) | right;
-      },
-    );
+  constructor(readonly signed: boolean) {
+    super();
+  }
+  _minSize = 16;
+  _encode(cursor: Cursor, value: bigint) {
+    cursor.view.setBigInt64(cursor.i, value, true);
+    cursor.view.setBigInt64(cursor.i + 8, value >> 64n, true);
+    cursor.i += 16;
+  }
+  _decode(cursor: Cursor) {
+    const right = cursor.view.getBigUint64(cursor.i, true);
+    const left = cursor.view[
+      this.signed ? "getBigInt64" : "getBigUint64"
```

Comment:
Should this be extracted into a member, assigned at construction?

Reply:
Yeah, one moment